components/project-card/project-card.jsx

```diff
@@ -76,18 +125,21 @@ class ProjectCard extends React.Component {
       });
     }
 
-    let bookmarks = getBookmarks();
-    let index;
+    ReactGA.event(this.createGaEventConfig(`Bookmark button`, this.state.bookmarked ? `Unbookmarked` : `Bookmarked`));
 
-    if (bookmarks) {
-      index = bookmarks.indexOf(this.props.id);
-      if (index > -1) {
-        this.unbookmarkProject(bookmarks,index);
-      } else {
-        this.bookmarkProject(bookmarks);
-      }
-      saveBookmarks(bookmarks);
+    if (user.loggedin) {
+      this.toggleBookmarkedStateOnDb((error) => {
```

Comment:
the client doesn't know anything about databases, it just knows about a server. That said, it is better to assume that "without any qualifiers", we're talking about the server so that `toggleBookmark` goes to the server, and `toggleBookmarkInLocalStorage` doesn't.